### PR TITLE
add uris

### PR DESCRIPTION
**Pull request by KeeyanGhoreshi**
## Describe your changes

Adds canonical uris to questionnaire's and uses them

## Issue ticket number and Jira link

[REMS-400](https://jira.mitre.org/browse/REMS-400)

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] Ensure the target / base branch for any feature PR is set to `dev` not main (the only exception to this is releases from `dev` and hotfix branches)

## Checklist for conducting a review
- [ ] Review the code changes and make sure they all make sense and are necessary. 
- [ ] Pull the PR branch locally and test by running through workflow and making sure everything works as it is supposed to.

### CODE REVIEW

**Review by jtquach1 — CHANGES_REQUESTED** [review]
The requested changes are not related to your ticket/PR, but they are related to the extensions your PR touches. The `Questionnaire.url` changes you made still conform to the `uri` type in FHIR R4. Ditto for `Questionnaire.item[sdc-questionnaire-subQuestionnaire].extension.valueCanonical`. 

I wasn't able to locate the original `Questionnaire.extension`s located at `http://hl7.org/fhir/StructureDefinition/sub-questionnaire` and `http://hl7.org/fhir/StructureDefinition/sub-questionnaire-expand`, but I found updated URLs for those extensions that conform to FHIR version 4.0.1 in the Structured Data Capture IG (mentioned under https://hl7.org/fhir/R4/questionnaire.html#scope). They seem to not be defined for FHIR version 4.0.0 for that IG, however. I was also unable to locate the sub-questionnaire and sub-questionnaire-expand extensions from https://hl7.org/fhir/R4/questionnaire.html. 

I'm not entirely sure whether `Questionnaire.item[sdc-questionnaire-collapsible].extension.valueCanonical` is meant to be a more updated version of `Questionnaire.item[sub-questionnaire-expand].extension.valueCanonical`, but its required `valueCode` ValueSet (not `valueBoolean`) is binary just like a boolean. This is the spec: https://hl7.org/fhir/uv/sdc/ValueSet-collapsible.html

File: src/fhir/questionnaireUtilities.ts [hunk 1]
```diff
@@ -186,7 +185,7 @@ export class QuestionnaireUtilities {
         if (expandExt && expandExt.valueBoolean) {
           expandRootItem = expandExt.valueBoolean;
         }
-        const subQuestionnaire = await this.findQuestionnaire(id);
+        const subQuestionnaire = await this.findQuestionnaireByUrl(subQ);
```
[the remark this answers is not in the document]
Comment (KeeyanGhoreshi) [reply src/fhir/questionnaireUtilities.ts:188]: not necessarily

[the remark this answers is not in the document]
Comment (jtquach1) [reply src/fhir/questionnaireUtilities.ts:188]: If the function needs to stay in here for future R4 (or R5) Questionnaire support for example, maybe we should leave a comment indicating that

**Review by avirgulto — APPROVED** [review]
Looks good

**Pull request merged**